Document heat transfer kernels. (#29646)
idaholab/moose: framework/doc/content/source/kernels/BodyForce.md

## Description

`BodyForce` implements a force term, such as a heat generation/sink term for heat
conduction, a momentum source/sink for momentum transport or structural mechanics, or
a source/sink term in species/mass transport. The context of this kernel depends
on the differential equation of interest, but shares the strong form on a domain
$\Omega$ as

\begin{equation}
\underbrace{-f}_{\textrm{BodyForce}} + \text{other kernels} = 0 \in \Omega
\end{equation}

> where $f$ is the source term (negative if a sink) and the second term on the

Reviewer: ```suggestion
where $f$ is the source term (negative if a sink) and "other kernels"
```

> left hand side represents the strong forms of other kernels which may be present in the equation. The `BodyForce`

Reviewer: ```suggestion
represent the strong forms of other terms present in the equation. The `BodyForce`
```
